### PR TITLE
theme: implement or fix xfailed tests

### PR DESCRIPTION
**Pull request by jacquerie**
Not yet fulfilling the promise in https://github.com/inspirehep/inspire-next/pull/932#issuecomment-189243461.

### CODE REVIEW

File: tests/unit/theme/test_theme_jinja2filters.py [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+    email_links, epoch_to_year_format, experiment_date, experiment_link,
+    format_cnum_with_hyphons, format_cnum_with_slash, format_date, has_space,
+    institutes_links, is_intbit_set, is_list, is_upper, join_array,
+    join_nested_lists, latex, limit_facet_elements, link_to_hep_affiliation,
+    long_timeout, new_line_after, number_of_records, number_of_search_results,
+    proceedings_link, publication_info, remove_duplicates,
+    remove_duplicates_from_dict, sanitize_arxiv_pdf, sanitize_collection_name,
+    search_for_experiments, sort_list_by_dict_val, split_author_name,
+    strip_leading_number_plot_caption, url_links, words, words_to_end
+)
```
Comment (jacquerie) [inline tests/unit/theme/test_theme_jinja2filters.py]: Should probably be replaced with `*`.

**Comment by jacquerie** [comment]
Now fulfilling the promise in https://github.com/inspirehep/inspire-next/pull/932#issuecomment-189243461. The bugs I found are documented with an `xfailed` test, which can be run with `pytest --runxfail tests/unit` to see the error. I also provided a succinct `reason` parameter to help understanding what the error is; if it's not enough, ping me somewhere and I'll try to remember what was happening : )

**Comment by jalavik** [comment]
Thanks for this. When will the xfailed bugs be addressed? Is that already in the works?

I am asking because some of them seem trivial to fix and could potentially be part of this PR unless it steps on someone else's work. E.g. intbitset we no longer need to use I think @kaplun  And the new search API just need to be adjusted etc.

**Comment by jacquerie** [comment]
> Is that already in the works?

Not by me, because in fact I wasn't sure if @ioannistsanaktsidis was already working on this file. But yes, some are trivial to fix.

**Comment by kaplun** [comment]
Indeed `intbitset`is not needed anymore.

**Comment by jacquerie** [comment]
Tests got submitted in another PR: https://github.com/inspirehep/inspire-next/pull/967. I will reuse this PR to do the proposed changes (`Query` API change, other minor stuff).

File: inspirehep/modules/theme/jinja2filters.py [hunk 2]
```diff
@@ -284,13 +287,16 @@ def proceedings_link(record):
     out = ''
     if not cnum:
         return out
-    search_result = Query("cnum:%s and 980__a:proceedings" % (cnum,)).\
-        search()
-    if len(search_result):
-        if len(search_result) > 1:
+
+    query = Query("cnum:%s and 980__a:proceedings" % (cnum,))
+    result = es.search(index='records-hep', body=query.body)
```
Comment (jacquerie) [inline inspirehep/modules/theme/jinja2filters.py]: Maybe I should remove this `index` argument.

File: inspirehep/modules/theme/jinja2filters.py [hunk 3]
```diff
@@ -399,9 +406,11 @@ def collection_select_current(collection_name, current_collection):
 @blueprint.app_template_filter()
 def number_of_records(collection_name):
     """Returns number of records for the collection."""
-    # return len(Query("").
-    #            search(collection=collection_name))
-    return 0
+    query = Query()
+    index = collection_to_index(collection_name)
+    result = es.count(index=index, body=query.body)
```
Comment (jacquerie) [inline inspirehep/modules/theme/jinja2filters.py]: This could be abstracted in a method.

File: inspirehep/modules/theme/jinja2filters.py [hunk 4]
```diff
@@ -510,15 +519,17 @@ def number_of_search_results(query, collection_name):
...
                 "number_of_hits"
             ]
-    # Replay the Query to get total number of hits
-    return len(Query('{query} AND collection:"{collection}"'.format(
-        {'query': query, 'collection': collection_name}
-    )))
+
+    query = Query(query)
+    index = collection_to_index(collection_name)
+    result = es.count(index=index, body=query.body)
```
Comment (jacquerie) [inline inspirehep/modules/theme/jinja2filters.py]: Similarly, this could be abstracted.

**Comment by jacquerie** [comment]
Most xfailed tests are now green, except two, who required more effort and were kept xfailed. I also decided to keep the querying concrete, so that we can abstract them when other people need that logic, so that the API will be as flexible as possible.

**Comment by jacquerie** [comment]
@jalavik Now incorporates your oral comments of yesterday.

**Pull request merged**